gdnative-sys/build.rs

```diff
@@ -337,26 +337,37 @@ mod api_wrapper {
             let v_min = api.version.minor;
             let gd_api_struct = api.godot_api_struct();
             godot_apis.extend(quote! {
-                let #i = find_api_ptr(core_api_struct, #gd_api_type, #v_maj, #v_min) as *const #gd_api_struct;
+                let #i = find_api_ptr(core_api_struct, #gd_api_type, #v_maj, #v_min)? as *const #gd_api_struct;
             });
             for function in &api.functions {
                 let function_name = function.rust_name();
-                let expect_msg = format!(
+                let message = format!(
                     "API function missing: {}.{}",
                     api.godot_api_struct(),
                     function_name
                 );
                 constructed_struct_fields.extend(quote! {
-                    #function_name: (*#i).#function_name.expect(#expect_msg),
+                    #function_name: Self::map_not_found((*#i).#function_name, #message)?,
                 });
             }
         }
         quote! {
-            pub unsafe fn from_raw(core_api_struct: *const godot_gdnative_core_api_struct) -> Self {
+            fn map_not_found<T>(t: Option<T>, message: &'static str) -> Result<T, InitError> {
+                match t {
+                    Some(t) => Ok(t),
+                    None => Err(InitError::NotFound{message: message.to_string()}),
+                }
+            }
```

Comment:
It seems like this would belong on the `InitError` type instead. Also, there is no need to convert the messages to owned `String`s if they are always `'static`: you can just make the message field `&'static str` in `InitError` instead.

Reply:
I moved it to the impl of `InitError`. However I'm not sure it belongs there; this is a helper function for `GodotApi::from_raw`, and the signature seems a bit specific to what was needed there.

The message field in `InitError` is not alway `'static`; in `find_api_ptr` I construct a message at runtime from a `GDNATIVE_API_TYPES` value.

Reply:
I see! I didn't notice that the message isn't always `'static`. It can just be a free function, then. It does not have to be associated with `GodotApi` (and certainly not in generator output in any case. Note that you can have multiple `impl`s on any type!)

Reply:
Changed to a free function.